src/components/themed/TransactionListTop.tsx

```diff
@@ -421,31 +420,32 @@ export class TransactionListTopComponent extends React.PureComponent<Props, Stat
       <>
         <View style={styles.balanceBoxWalletNameCurrencyContainer}>
           <EdgeTouchableOpacity accessible={false} style={styles.balanceBoxWalletNameContainer} onPress={this.handleOpenWalletListModal}>
-            <CryptoIconUi4 sizeRem={1.5} tokenId={tokenId} walletId={wallet.id} />
             <EdgeText accessible style={styles.balanceBoxWalletName}>
               {walletName}
             </EdgeText>
-            <Ionicons name="chevron-forward" size={theme.rem(1.5)} color={theme.iconTappable} />
+            <View style={styles.balanceBoxChevronAdjust}>
+              <Ionicons name="chevron-down" size={theme.rem(1.5)} color={theme.iconTappable} />
+            </View>
           </EdgeTouchableOpacity>
-          <EdgeTouchableOpacity testID="gearIcon" onPress={this.handleMenu} style={styles.settingsIcon}>
-            <Fontello accessibilityHint={lstrings.wallet_settings_label} color={theme.iconTappable} name="control-panel-settings" size={theme.rem(1.5)} />
+          <EdgeTouchableOpacity testID="gearIcon" onPress={this.handleMenu} style={styles.settingsTouchContainer}>
+            <Entypo accessibilityHint={lstrings.wallet_settings_label} color={theme.icon} name="dots-three-vertical" size={theme.rem(1.25)} />
           </EdgeTouchableOpacity>
         </View>
         <EdgeTouchableOpacity accessible={false} onPress={this.props.toggleBalanceVisibility}>
-          {isAccountBalanceVisible ? (
-            <>
-              <EdgeText accessible style={styles.balanceBoxCurrency} minimumFontScale={0.3}>
-                {cryptoAmountFormat + ' ' + displayDenomination.name}
-              </EdgeText>
-              <EdgeText accessible style={styles.balanceFiatBalance}>
-                {fiatSymbol + fiatBalanceFormat + ' ' + fiatCurrencyCode}
-              </EdgeText>
-            </>
-          ) : (
-            <EdgeText accessible style={styles.balanceFiatShow}>
-              {lstrings.string_show_balance}
+          <View style={styles.balanceBoxCryptoBalanceContainer}>
+            <EdgeText accessible style={styles.balanceBoxCurrency} minimumFontScale={0.3}>
+              {(isAccountBalanceVisible ? cryptoAmountFormat : lstrings.redacted_placeholder) + ' ' + displayDenomination.name}
             </EdgeText>
-          )}
+            <Ionicons
+              name={isAccountBalanceVisible ? 'eye-off-outline' : 'eye-outline'}
+              style={styles.eyeIcon}
+              color={theme.iconTappable}
+              size={theme.rem(2)}
```

Comment:
2 rem looks huge. I think the design probably meant 1.75. Although I tried 1.5 and that looks much better

<img width="445" alt="Screenshot 2024-03-26 at 17 40 04" src="https://github.com/EdgeApp/edge-react-gui/assets/6079354/abb74c2e-247c-48c9-be29-f0f4c6765312">

Reply:
Also, the amount will have to wrap to 2 lines in absolutely worst case such as this one

<img width="438" alt="Screenshot 2024-03-26 at 17 42 53" src="https://github.com/EdgeApp/edge-react-gui/assets/6079354/f199fe94-e174-4b36-ba66-283faed37f1b">

Reply:
Or you need to ensure the icon doesn't exceed the bounding area and try to get the text to shrink further

Reply:
![image](https://github.com/EdgeApp/edge-react-gui/assets/90650827/6ad9fcfb-546e-4779-99fa-ea15e3a258dd)

Reply:
The best solution would be to update `CryptoText`/`CryptoAmount` to handle these edge cases more gracefully. I think the best solution really is truncation, like we do in market cap/volume in the Markets scene

Reply:
Updated screens in the main description to show edge cases with the scaled eye icon and center align